composition/src/normalization/normalization-factory.ts

```diff
@@ -1352,32 +1363,61 @@ export class NormalizationFactory {
               isValid = false;
               continue;
             }
-            if (missingRequiredFieldNames.has(fieldName)) {
-              missingRequiredFieldNames.delete(fieldName);
+            if (allowedFieldNames.has(fieldName)) {
+              allowedFieldNames.delete(fieldName);
             } else {
-              duplicateRequiredFieldNames.add(fieldName);
+              duplicateFieldNames.add(fieldName);
               isValid = false;
               continue;
             }
-            if (field.value.kind !== Kind.STRING || field.value.value.length < 1) {
-              invalidRequiredFieldNames.add(fieldName);
-              isValid = false;
-              continue;
+            if (missingRequiredFieldNames.has(fieldName)) {
+              missingRequiredFieldNames.delete(fieldName);
             }
             switch (fieldName) {
               case CONSUMER_NAME:
+                if (field.value.kind != Kind.STRING || field.value.value.length < 1) {
+                  invalidRequiredFieldNames.add(fieldName);
+                  isValid = false;
+                  continue;
+                }
                 consumerName = field.value.value;
                 break;
               case STREAM_NAME:
+                if (field.value.kind != Kind.STRING || field.value.value.length < 1) {
+                  invalidRequiredFieldNames.add(fieldName);
+                  isValid = false;
+                  continue;
+                }
                 streamName = field.value.value;
                 break;
+              case CONSUMER_INACTIVE_THRESHOLD:
+                if (field.value.kind != Kind.INT) {
+                  errorMessages.push(
+                    invalidIntegerTypeValue('edfs__NatsStreamConfiguration(consumerInactiveThreshold: ...)'),
```

Comment:
Did you do it like this because some value kinds don't define value.value?

Reply:
I want them to be Int and not strings, when parsed. Also some kind has a value that is not a string.